src/sss_client/common.c

```diff
@@ -1144,7 +1128,26 @@ errno_t sss_strnlen(const char *str, size_t maxlen, size_t *len)
 }
 
 #if HAVE_PTHREAD
-typedef void (*sss_mutex_init)(void);
+bool sss_is_lockfree_mode(void)
+{
+    const char *env = NULL;
+    static enum {
+            MODE_UNDEF,
+            MODE_LOCKING,
+            MODE_LOCKFREE
+        } mode = MODE_UNDEF;
+
+    if (mode == MODE_UNDEF) {
+        env = getenv("SSS_NSS_LOCKFREE");
+        if ((env != NULL) && (strcmp(env, "NO") == 0)) {
+            mode = MODE_LOCKING;
+        } else {
+            mode = MODE_LOCKFREE;
+        }
+    }
+
+    return (mode == MODE_LOCKFREE);
+}
```

Comment:
`mode` is static on purpose - I want to read it once and to avoid execution of `getenv()` + `strcmp()` every client call.
This is not only because of performance reasons, but also because we don't want to support change of this var in runtime.

Reply:
Done.